bpf/flows.c

```diff
@@ -78,6 +78,20 @@ static inline void update_dns(additional_metrics *extra_metrics, pkt_info *pkt,
     }
 }
 
+static inline void add_observed_intf(additional_metrics *value, u32 if_index, u8 direction) {
+    if (value->nb_observed_intf < MAX_OBSERVED_INTERFACES) {
+        for (u8 i = 0; i < value->nb_observed_intf; i++) {
+            if (value->observed_intf[i].if_index == if_index &&
+                value->observed_intf[i].direction == direction) {
+                return;
+            }
+        }
+        value->observed_intf[value->nb_observed_intf].if_index = if_index;
+        value->observed_intf[value->nb_observed_intf].direction = direction;
+        value->nb_observed_intf++;
+    }
```

Comment:
we need an else here and log a msg and probably updated a counter I assume the value of 4 is based on what we see today but depends on how often we come to else we might need to increase it.

what happens when we hit the above condition is that a bad flow that we should drop or u can return error code in this function and do the print and counter update at the caller but we can't just be silent when this condition happens

Reply:
right, +1

Reply:
I think there's the same thing with MAX_NETWORK_EVENTS that we may silently ignore

Reply:
network events I return error so that should be fine

Reply:
Oh I didn't notice the rolling increase [here](https://github.com/netobserv/netobserv-ebpf-agent/blob/main/bpf/network_events_monitoring.h#L45)
So when capacity is reached, it erases the oldest entry, but it do so silently, or am I missing something?

Reply:
yeah that was by design newer events will overwrite older ones but as of now we can't get more than 3 events max per flow so we should be fine for now

Reply:
done here https://github.com/netobserv/netobserv-ebpf-agent/pull/470/commits/f2a7a94fe1b7bbf9791ebbe964ca5e13e7d7caea